src/app/screens/developer/components/timeentry/timeentry.component.ts

```diff
@@ -0,0 +1,147 @@
+import { Component, OnInit } from '@angular/core';
+import { MatTableDataSource } from '@angular/material/table';
+import { FormGroup, FormControl } from '@angular/forms';
+
+import { ProjectloggingService } from 'src/app/services/projectlogging.service';
+import { ProjectService } from 'src/app/services/project.service';
+
+
+export interface Data {
+  title: string;
+ 
+}
+const ELEMENT_DATA: Data[] = [
+  {
+    title: 'Projects',
+  }
+];
+
+@Component({
+  selector: 'app-timeentry',
+  templateUrl: './timeentry.component.html',
+  styleUrls: ['./timeentry.component.css']
+})
+export class TimeentryComponent implements OnInit {
+
+  incomingProject = { project_id: 2, title: '' };
+  showAlert: boolean = false;
+  userProjects: any
+
+  hourform = new FormGroup({
+    project_hours: new FormControl(''),
+    volunteering_hours: new FormControl(''),
+    vacation: new FormControl(''),
+    sick: new FormControl(''),
+    emp_id: new FormControl(),
+    project_id: new FormControl(),
+    logged_date: new FormControl(new Date().toISOString().slice(0, 10)),
+  });
+  constructor(
+    private plog: ProjectloggingService,
+    private projectService: ProjectService
+  ) {}
+  newProject = '';
+  assignedprojects =[{title: "tpms"}]
+  displayedColumns: string[] = [
+    'Projects',
+    'Volunteering',
+    'Sick',
+    'Vacation'
+  ];
+  dataSource = new MatTableDataSource(ELEMENT_DATA);
+  dummy = [];
+
+  applyFilter(filterValue: string) {
+    this.dataSource.filter = filterValue.trim().toLowerCase();
+  }
+
+  ngOnInit() {
+    
+    var userData = JSON.parse(localStorage.getItem("userProjects"))
+    console.log("User Data | ",userData);
+    
+    if(userData === null){
+      this.userProjects = [];
+    }else{
+      this.userProjects = userData;
+    }
+    
+  }
+
+  sickFieldChecked(event) {
+    if (event == 'sick') {
+      this.hourform.controls.vacation.disable();
+      this.hourform.controls.volunteering_hours.disable();
+      this.hourform.controls.project_hours.disable();
+    }
+    else {
+      this.hourform.controls.vacation.enable();
+      this.hourform.controls.volunteering_hours.enable();
+      this.hourform.controls.project_hours.enable();
+    }
+  }
+
+  vacationFieldChecked(event) {
+    if (event == 'vacation') {
+      this.hourform.controls.sick.disable();
+      this.hourform.controls.volunteering_hours.disable();
+      this.hourform.controls.project_hours.disable();
+    }
+    else {
+      this.hourform.controls.sick.enable();
+      this.hourform.controls.volunteering_hours.enable();
+      this.hourform.controls.project_hours.enable();
+    }
+  }
+
+  submitChecked(event: string) {
+    if (event === 'true') {
```

Comment:
why is event true here, it's 'vacation' or 'sick' elsewhere? What triggers this function? all it does is reset fields, so why isn't it called resetFields? When would 'event' ever be false?